### PR TITLE
Update DG to include details of implementation of Assign command

### PR DESCRIPTION
**Pull request by cyiting**

### CODE REVIEW

**Comment by Guo-KeCheng** [comment]
Yiting, could you remove 3.1 and 3.2 under DG? Then we get nicholas to repush his part to get the system to log his contributions.

File: docs/images/AssignActivityDiagram.png
Comment (kayyenl) [inline docs/images/AssignActivityDiagram.png:1]: Besides the PlantUML constraints, it is good! Just the right amount of detail for the user. 👍

**Comment by kayyenl** [comment]
LGTM!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -360,21 +360,62 @@ Purpose: Allow users to edit tasks that are currently listed in OfficeConnect.
...
+#### 3.6.2 Design Considerations
+**Aspect: Steps Leading to Execution of Assign Command**
+
+* **Alternative 1 (current choice):** Users execute `list` or `find` to display a list of persons and `listtask` or
+  `findtask` to display a list of tasks on the application.
+  * Pros: Users can search for a specific person or task before assigning.
+  * Cons: If `list` and `listtask` are used, users may have to scroll through a long list to search for the desired
+    person or task. The results of `find` and `findtask` cannot be displayed simultaneously. Thus, if `find` was executed
+    to search for a person, the user will then have to execute `listtask` to access the full list of tasks in order to
+    identify the task that the user wants to be assigned to that person.
```
Comment (SPWwj) [inline docs/DeveloperGuide.md:396]: You are sharp, we can discuss how can we do better in v1.3b

**Review by SPWwj — APPROVED** [review]
LGTM

**Pull request merged**